nvflare/private/fed/client/admin.py

```diff
@@ -157,3 +166,17 @@ def _dispatch_request(
         else:
             reply = error_reply("invalid_request")
         return new_cell_message({}, reply)
+
+    def _set_security_data(self, engine, req):
+        security_items = {}
+        with engine.new_context() as fl_ctx:
+            security_items[FLContextKey.USER_NAME] = req.get_header(RequestHeader.USER_NAME, "")
+            security_items[FLContextKey.USER_ORG] = req.get_header(RequestHeader.USER_ORG, "")
+            security_items[FLContextKey.USER_ROLE] = req.get_header(RequestHeader.USER_ROLE, "")
+            security_items[FLContextKey.SUBMITTER_NAME] = req.get_header(RequestHeader.SUBMITTER_NAME, "")
+            security_items[FLContextKey.SUBMITTER_ORG] = req.get_header(RequestHeader.SUBMITTER_ORG, "")
+            security_items[FLContextKey.SUBMITTER_ROLE] = req.get_header(RequestHeader.SUBMITTER_ROLE, "")
+
+            security_items[FLContextKey.JOB_META] = req.get_header(RequestHeader.JOB_META, {})
+
+            fl_ctx.set_prop(FLContextKey.SECURITY_ITEMS, security_items, private=True, sticky=True)
```

Comment:
this code is duplicated, can we make a method and reuse that?

Reply:
These are not exactly the same. One for getting from the Connection, another one from the Message. (Two different usage scenario.)